internal/provider/job_resource_test.go

```diff
@@ -99,6 +99,16 @@ func TestJobResourceCreateRequestBody(t *testing.T) {
 			},
 			expected: []byte(`{"inventory": 3}`),
 		},
+		{
+			name: "wait_for_completed parameters",
+			input: JobResourceModel{
+				InventoryID:              basetypes.NewInt64Value(3),
+				TemplateID:               types.Int64Value(1),
+				WaitForCompletion:        basetypes.NewBoolValue(true),
+				WaitForCompletionTimeout: basetypes.NewInt64Value(60),
+			},
+			expected: []byte(`{"inventory":3}`),
+		},
```

Comment:
Glad to see the new fields are tested here. I think [acceptance tests](https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests) will be needed to cover most of the functionality though (e.g. that a job resource created with this flag is not marked as 'created' by terraform until the job reaches one of these final states.

We can add that in a follow up PR.